sdks/python/apache_beam/runners/direct/evaluation_context.py

```diff
@@ -44,6 +44,10 @@ def get_step_context(self):
       self._step_context = DirectStepContext(self.keyed_states)
     return self._step_context
 
+  def reset(self):
+    if self._step_context:
```

Comment:
<!--new_thread; commit:2bc3b2426c278088706b430164375d35dc5cd148; resolved:0-->
You don't need the if here.